src/wireframes/model/projections.ts

```diff
@@ -129,6 +135,41 @@ export const getSelectedConfigurables = createSelector(
     shape => (shape ? shape.configurables : EMPTY_CONFIGURABLES),
 );
 
+
+export const getDocumentColors = createSelector(
+    getEditorDiagramItems,
+    (items) => {
+        const elements = Object.values(items) as any[];
+        const result = new Set();
+
+        for (const element of elements) {
+            for (let key in element.values.items.items) {
+            const backgroundColor =
+            element.values.items.items[key].values.appearance.items
+                .BACKGROUND_COLOR;
+                if (typeof backgroundColor === 'string' || backgroundColor instanceof String){
```

Comment:
Shapes can add their own appearance properties: https://github.com/mydraft-cc/ui/blob/master/src/wireframes/shapes/neutral/horizontal-scrollbar.ts#L43

So every property can be a color. The name does not really matter. A value is a color if (value.length === 7 && value[0] == '#')